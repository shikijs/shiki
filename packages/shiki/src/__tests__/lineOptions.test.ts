import { getHighlighter } from '../index'

test('applies class to a single line', async () => {
  const highlighter = await getHighlighter({
    theme: 'css-variables',
    langs: ['js']
  })

  const code = `
console.log('line 1');
console.log('line 2');
`.trim()

  const html = highlighter.codeToHtml(code, {
    lang: 'js',
    lineOptions: [{ line: 1, classes: ['highlighted'] }]
  })
  expect(html).toMatchSnapshot()
})

test('splits plaintext into lines', async () => {
  const highlighter = await getHighlighter({
    theme: 'css-variables'
  })

  const code = `
lorem ipsum
dolor sit amet
`.trim()

  const html = highlighter.codeToHtml(code, {
    lang: 'txt',
    lineOptions: [{ line: 1, classes: ['highlighted'] }]
  })
  expect(html).toMatchSnapshot()
})

test('applies different classes to different lines', async () => {
  const highlighter = await getHighlighter({
    theme: 'css-variables',
    langs: ['js']
  })

  const code = `
console.log('line 1');
console.log('line 2');
console.log('line 3');
console.log('line 4');
console.log('line 5');
`.trim()

  const html = highlighter.codeToHtml(code, {
    lang: 'js',
    lineOptions: [
      { line: 1, classes: ['class-foo'] },
      { line: 3, classes: ['class-bar'] },
      { line: 4, classes: ['class-baz'] }
    ]
  })
  expect(html).toMatchSnapshot()
})

test('deduplicates classes per line', async () => {
  const highlighter = await getHighlighter({
    theme: 'css-variables',
    langs: ['js']
  })

  const code = `
console.log('line 1');
console.log('line 2');
`.trim()

  const html = highlighter.codeToHtml(code, {
    lang: 'js',
    lineOptions: [
      { line: 1, classes: ['highlighted'] },
      { line: 1, classes: ['highlighted'] }
    ]
  })
  expect(html).toMatchSnapshot()
})

test('ignores line numbers out of range', async () => {
  const highlighter = await getHighlighter({
    theme: 'css-variables',
    langs: ['js']
  })

  const code = `
console.log('line 1');
console.log('line 2');
console.log('line 3');
`.trim()

  const html = highlighter.codeToHtml(code, {
    lang: 'js',
    lineOptions: [
      { line: -1, classes: ['highlighted'] },
      { line: 0, classes: ['highlighted'] },
      { line: 2, classes: ['highlighted'] },
      { line: 4, classes: ['highlighted'] },
      { line: 5, classes: ['highlighted'] }
    ]
  })
  expect(html).toMatchSnapshot()
})
