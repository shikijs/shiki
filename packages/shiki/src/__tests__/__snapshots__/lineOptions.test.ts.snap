// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`applies class to a single line 1`] = `
"<pre class=\\"shiki\\" style=\\"background-color: var(--shiki-color-background)\\"><code><span class=\\"line highlighted\\"><span style=\\"color: var(--shiki-token-constant)\\">console</span><span style=\\"color: var(--shiki-token-function)\\">.log</span><span style=\\"color: var(--shiki-color-text)\\">(</span><span style=\\"color: var(--shiki-token-string-expression)\\">&#39;line 1&#39;</span><span style=\\"color: var(--shiki-color-text)\\">);</span></span>
<span class=\\"line\\"><span style=\\"color: var(--shiki-token-constant)\\">console</span><span style=\\"color: var(--shiki-token-function)\\">.log</span><span style=\\"color: var(--shiki-color-text)\\">(</span><span style=\\"color: var(--shiki-token-string-expression)\\">&#39;line 2&#39;</span><span style=\\"color: var(--shiki-color-text)\\">);</span></span></code></pre>"
`;

exports[`applies different classes to different lines 1`] = `
"<pre class=\\"shiki\\" style=\\"background-color: var(--shiki-color-background)\\"><code><span class=\\"line class-foo\\"><span style=\\"color: var(--shiki-token-constant)\\">console</span><span style=\\"color: var(--shiki-token-function)\\">.log</span><span style=\\"color: var(--shiki-color-text)\\">(</span><span style=\\"color: var(--shiki-token-string-expression)\\">&#39;line 1&#39;</span><span style=\\"color: var(--shiki-color-text)\\">);</span></span>
<span class=\\"line\\"><span style=\\"color: var(--shiki-token-constant)\\">console</span><span style=\\"color: var(--shiki-token-function)\\">.log</span><span style=\\"color: var(--shiki-color-text)\\">(</span><span style=\\"color: var(--shiki-token-string-expression)\\">&#39;line 2&#39;</span><span style=\\"color: var(--shiki-color-text)\\">);</span></span>
<span class=\\"line class-bar\\"><span style=\\"color: var(--shiki-token-constant)\\">console</span><span style=\\"color: var(--shiki-token-function)\\">.log</span><span style=\\"color: var(--shiki-color-text)\\">(</span><span style=\\"color: var(--shiki-token-string-expression)\\">&#39;line 3&#39;</span><span style=\\"color: var(--shiki-color-text)\\">);</span></span>
<span class=\\"line class-baz\\"><span style=\\"color: var(--shiki-token-constant)\\">console</span><span style=\\"color: var(--shiki-token-function)\\">.log</span><span style=\\"color: var(--shiki-color-text)\\">(</span><span style=\\"color: var(--shiki-token-string-expression)\\">&#39;line 4&#39;</span><span style=\\"color: var(--shiki-color-text)\\">);</span></span>
<span class=\\"line\\"><span style=\\"color: var(--shiki-token-constant)\\">console</span><span style=\\"color: var(--shiki-token-function)\\">.log</span><span style=\\"color: var(--shiki-color-text)\\">(</span><span style=\\"color: var(--shiki-token-string-expression)\\">&#39;line 5&#39;</span><span style=\\"color: var(--shiki-color-text)\\">);</span></span></code></pre>"
`;

exports[`deduplicates classes per line 1`] = `
"<pre class=\\"shiki\\" style=\\"background-color: var(--shiki-color-background)\\"><code><span class=\\"line highlighted\\"><span style=\\"color: var(--shiki-token-constant)\\">console</span><span style=\\"color: var(--shiki-token-function)\\">.log</span><span style=\\"color: var(--shiki-color-text)\\">(</span><span style=\\"color: var(--shiki-token-string-expression)\\">&#39;line 1&#39;</span><span style=\\"color: var(--shiki-color-text)\\">);</span></span>
<span class=\\"line\\"><span style=\\"color: var(--shiki-token-constant)\\">console</span><span style=\\"color: var(--shiki-token-function)\\">.log</span><span style=\\"color: var(--shiki-color-text)\\">(</span><span style=\\"color: var(--shiki-token-string-expression)\\">&#39;line 2&#39;</span><span style=\\"color: var(--shiki-color-text)\\">);</span></span></code></pre>"
`;

exports[`ignores line numbers out of range 1`] = `
"<pre class=\\"shiki\\" style=\\"background-color: var(--shiki-color-background)\\"><code><span class=\\"line\\"><span style=\\"color: var(--shiki-token-constant)\\">console</span><span style=\\"color: var(--shiki-token-function)\\">.log</span><span style=\\"color: var(--shiki-color-text)\\">(</span><span style=\\"color: var(--shiki-token-string-expression)\\">&#39;line 1&#39;</span><span style=\\"color: var(--shiki-color-text)\\">);</span></span>
<span class=\\"line highlighted\\"><span style=\\"color: var(--shiki-token-constant)\\">console</span><span style=\\"color: var(--shiki-token-function)\\">.log</span><span style=\\"color: var(--shiki-color-text)\\">(</span><span style=\\"color: var(--shiki-token-string-expression)\\">&#39;line 2&#39;</span><span style=\\"color: var(--shiki-color-text)\\">);</span></span>
<span class=\\"line\\"><span style=\\"color: var(--shiki-token-constant)\\">console</span><span style=\\"color: var(--shiki-token-function)\\">.log</span><span style=\\"color: var(--shiki-color-text)\\">(</span><span style=\\"color: var(--shiki-token-string-expression)\\">&#39;line 3&#39;</span><span style=\\"color: var(--shiki-color-text)\\">);</span></span></code></pre>"
`;

exports[`splits plaintext into lines 1`] = `
"<pre class=\\"shiki\\" style=\\"background-color: var(--shiki-color-background)\\"><code><span class=\\"line highlighted\\"><span style=\\"color: var(--shiki-color-text)\\">lorem ipsum</span></span>
<span class=\\"line\\"><span style=\\"color: var(--shiki-color-text)\\">dolor sit amet</span></span></code></pre>"
`;
